### PR TITLE
Fix/npm publish

### PR DESCRIPTION
**Pull request by linhlecle**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,12 +20,14 @@
...
   ],
   "main": "lib/index.js",
-  "types": "lib/index",
+  "typings": "lib/index.d.ts",
   "scripts": {
     "clean": "rimraf lib && rimraf coverage",
     "format": "yarn prettier --write '{src,__tests__}/**/*.ts'",
     "lint": "tslint -p . --fix",
     "prebuild": "yarn clean",
+    "prepare": "npm run build",
```
Comment (manh-vv) [inline package.json]: @linhlecle I think you should not add this script because we have built before publish already.

File: package.json [hunk 2]
```diff
@@ -20,12 +20,14 @@
     "lib"
   ],
   "main": "lib/index.js",
-  "types": "lib/index",
+  "typings": "lib/index.d.ts",
```
Comment (manh-vv) [inline package.json]: Do we really need to change from `types` to `typings`?

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

File: package.json [hunk 3]
```diff
@@ -20,12 +20,14 @@
...
   "scripts": {
     "clean": "rimraf lib && rimraf coverage",
     "format": "yarn prettier --write '{src,__tests__}/**/*.ts'",
     "lint": "tslint -p . --fix",
     "prebuild": "yarn clean",
+    "prepare": "npm run build",
+    "prepublish": "npm run build",
```
Comment (manh-vv) [inline package.json]: pls use yarn as default: `yarn build`

Comment (linhlecle) [reply package.json]: @manh-vv  I know. But how to user use the npm package when they don't have yarn.

**Pull request merged**